### PR TITLE
Cleanup TESTING and DEVELOPER_GUIDE markdowns

### LINKED ISSUE #943: [BUG] Updates needed in DEVELOPER_GUIDE.md, TESTING.md, etc.
**Describe the bug**
There are some documentation issues that need to be addressed in the repo docs. 

- [x] In DEVELOPER_GUIDE.md under the heading 'Project Layout', it mentions that 'docs' contains the documentation for the project. This is incorrect as that folder no longer exists.
- [x] In CONTRIBUTING.md, "please see OpenSearch Docs repo." points to README.md but I think it should point to [here](https://github.com/opensearch-project/documentation-website) since the docs folder no longer exists. 
- [ ] The MAINTAINERS.md lists 13 maintainers and points to [this](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) file which lists 2. If there is a difference we should make it clear, if not we should should correct. 
- [x] In TESTING.md under the heading "Testing Backwards Compatibility" it references x-pack in an example e.g. "specific tests from the x-pack rolling upgrade from 7.7.0".
- [ ] RELEASING.md has blank/empty TODO at the bottom. 
- [x] In TESTING.md under the heading "Testing Backwards Compatibility" it references the link to VersionCollection which is 404ing.
- [x] In TESTING.md under the heading "Refactor code to make it easier to test" it references an Elastic issue. We should remove or update to an OpenSearch example. 
- [x]  In TESTING.md under the heading "Test coverage analysis" it says "Test coverage reporting used to be possible with JaCoCo when OpenSearch was using Maven as its build system" but this is inaccurate. OpenSearch never used maven. This paragraph also mentions an elastic issue. I would suggest dropping the whole paragraph.
- [x] We lowered the requirements for Java from 14 to 11. This [doc](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#requirements) should be updated.
- [x] In TESTING.md under the heading "Building with extra plugins" 'opensearch-extra' should be formatted as inline code.
- [x] In DEVELOPER_GUIDE.md under the heading 'Project Layout', it mentions that 'plugins' contains the 'Officially supported plugins to OpenSearch' - while it might be meaningful for ODFE plugins, it's confusing for OpenSearch. 
- [x] In client/benchmark/README.md references a document on benchmarks.elasticsearch.org.s3.amazonaws.com.

Lastly, TESTING.md references Rally - I'm not sure that OpenSearch will work with Rally and we should change this to OpenSearch Benchmarks as soon as it is released. My suggestion would be to drop the benchmarking section until OpenSearch Benchmarks is out.

**To Reproduce**
Open repo docs. 

**Expected behavior**
Listed above

**Plugins**
N/A

**Screenshots**
N/A

**Host/Environment (please complete the following information):**
 - OS: N/A
 - Version: N/A

**Additional context**
None.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Checking the boxes in https://github.com/opensearch-project/OpenSearch/issues/943
 
### Issues Resolved

Closes #943.
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
start gradle check

File: TESTING.md [hunk 1]
```diff
@@ -34,15 +34,14 @@ OpenSearch uses [jUnit](https://junit.org/junit5/) for testing, it also uses ran
...
-- [Benchmarking](#benchmarking)
 
 # Requirements
 
 You will need the following pieces of software to run these tests:
 
 - Docker & Docker Compose
 - Vagrant
-- JDK 14
+- JDK 11
```
Comment (skkosuri-amzn) [inline TESTING.md:44]: Are we using 11 for the GA artifacts build?

Comment (dblock) [reply TESTING.md:44]: No, not for GA. This is on main and see https://github.com/opensearch-project/OpenSearch/pull/636.

**Review by skkosuri-amzn — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -175,7 +170,7 @@ For example, reindex requires the `connect` permission so it can perform reindex
 
 ### `plugins`
 
-Officially supported plugins to OpenSearch. We decide that a feature should be a plugin rather than shipped as a module because we feel that it is only important to a subset of users, especially if it requires extra dependencies.
+Built-in plugins to OpenSearch. We decide that a feature should be a plugin rather than shipped as a module because we feel that it is only important to a subset of users, especially if it requires extra dependencies.
```
Comment (adnapibar) [inline DEVELOPER_GUIDE.md:173]: Correct me if I'm wrong but I believe these plugins not built-in into the distributions and users have to install these manually.

Comment (dblock) [reply DEVELOPER_GUIDE.md]: You're right. Not sure why I changed this, reverting.

**Comment by dblock** [comment]
start gradle check

**Review by adnapibar — APPROVED** [review]
LGTM

**Comment by VachaShah** [comment]
Hi @dblock, I see the issue #943 has the last item to update the `elasticsearch` url for benchmarks. Is there a reason we have not updated it? I am curious since I have to fix a [broken link](https://github.com/opensearch-project/OpenSearch/tree/main/client/benchmark#bulk-indexing) in that file as part of issue #894 and was wondering if there is a conclusion on what link we will be using.

**Comment by dblock** [comment]
> Hi @dblock, I see the issue #943 has the last item to update the `elasticsearch` url for benchmarks. Is there a reason we have not updated it? I am curious since I have to fix a [broken link](https://github.com/opensearch-project/OpenSearch/tree/main/client/benchmark#bulk-indexing) in that file as part of issue #894 and was wondering if there is a conclusion on what link we will be using.

I just followed the recommendation to remove that section. Not sure what the link should be, feel free to put the section back with the correct link when you figure it out!

**Pull request merged**